### PR TITLE
Feature/alarm kits sale

### PR DESCRIPTION
**Pull request by chrisdavery**
Tagged product: https://konnected.io/products/konnected-alarm-panel-pro-12-zone-kit?preview_theme_id=161737408827
Category: https://konnected.io/collections/partner-pricing-sample?preview_theme_id=161737408827

### CODE REVIEW

File: snippets/card-product.liquid [hunk 1]
```diff
@@ -118,7 +118,33 @@
...
+            {% assign tagSplit = tag | split: ':' %}
+            {% assign pTag = tagSplit.first %}
+            {% assign pLabel = tagSplit.last %}
+
+            {% assign hasTag = card_product.tags | where: pTag %}
+
+            {% if hasTag.size > 0 %}
+              {% assign tagLabel = pLabel %}
+            {% endif %}
+          {% endfor %}
```
Comment (heythisisnate) [inline snippets/card-product.liquid:136]: I don't think all of this is necessary. We don't need this complex customTag structure. The "partner pricing" badge shows up if product.type == 'Wholesale'.

File: snippets/card-product.liquid [hunk 2]
```diff
@@ -118,7 +118,33 @@
...
+          <div class="card__badge {{ settings.badge_position }}" {% if tagLabel != 0 %} data-tag="{{ tagLabel }}"{% endif %}>
+            {% if tagLabel != 0 %}
+              <span
+                class="badge custom-badge badge--bottom-left color-{{ settings.sale_badge_color_scheme }}"
+                style="--color-background: {{ settings.custom_badge_color_bg.red }}, {{ settings.custom_badge_color_bg.green }}, {{ settings.custom_badge_color_bg.blue }};"
+              >
+                {{- tagLabel -}}
+              </span>
+            {% endif %}
+
```
Comment (heythisisnate) [inline snippets/card-product.liquid:147]: custom badge colors not needed here either, just re-use the `sale_badge_color_scheme`

Comment (heythisisnate) [reply snippets/card-product.liquid:147]: We should keep with the stock badge. All we need is an additional `{%- if card_product.type == 'Wholesale' -%}` condition in this if/else block where it generates the badge text.

File: templates/collection.alarm-kits-sale.json
Comment (heythisisnate) [inline templates/collection.alarm-kits-sale.json:1]: Where did this new collection template come from? Why do we need it?
This should just apply on the default collection.json template

File: snippets/price.liquid [hunk 3]
```diff
@@ -30,13 +30,30 @@
...
+  {% assign pTag = tagSplit.first | strip %}
+  {% assign pLabel = tagSplit.last | strip %}
+
+  {% assign hasTag = product.tags | where: pTag %}
+
+  {% if hasTag.size > 0 %}
+    {% assign tagLabel = pLabel %}
+  {% endif %}
+{% endfor %}
+
```
Comment (heythisisnate) [inline snippets/price.liquid:48]: This is not needed. Should be as simple as:
```
{%- liquid
if product.type == 'Wholesale'
  show_badges = true
endif
-%}

then below where `show_badges` is used just add a little bit of code to change the badge to "Partner price" instead of "sale"

**Review by heythisisnate — CHANGES_REQUESTED** [review]
Reduce complexity as noted. Re-using the built-in "Sale" badge instead of adding a new custom badge

File: templates/collection.alarm-kits-sale.json
Comment (chrisdavery) [reply templates/collection.alarm-kits-sale.json:1]: need to keep this, mockup has a custom banner compared to collection.json

File: assets/konnected.css [hunk 4]
```diff
@@ -856,6 +856,16 @@ iframe#bwpFrame {
...
+.custom-badge {
+  color: rgb(var(--color-base-text));
+  font-weight: 600;
+  --badge-corner-radius: 0;
+}
+
+.has-cbadge .custom-badge {
+  display: inline-block !important;
+}
+
```
Comment (heythisisnate) [inline assets/konnected.css:868]: don't need custom style
badge styles should be consistent (see thread: https://konnected-io.slack.com/archives/C046E5GMTCK/p1696347140664139)

File: config/settings_data.json [hunk 5]
```diff
@@ -17,6 +17,7 @@
     "card_style": "standard",
     "text_boxes_border_thickness": 0,
     "media_radius": 20,
+    "custom_badge_color_bg": "#47f5a3",
```
Comment (heythisisnate) [inline config/settings_data.json:20]: remove

File: config/settings_data.json [hunk 6]
```diff
@@ -40,6 +41,7 @@
     "checkout_accent_color": "#3f75ff",
     "checkout_button_color": "#3f75ff",
     "checkout_error_color": "#ff6d6d",
+    "custom_badges": "partner-pricing: Partner Pricing",
```
Comment (heythisisnate) [inline config/settings_data.json:44]: remove

File: config/settings_schema.json [hunk 7]
```diff
@@ -1345,6 +1345,11 @@
         ],
         "default": "inverse",
         "label": "t:settings_schema.badges.settings.sold_out_badge_color_scheme.label"
+      },
+      {
+        "type": "color",
+        "id": "custom_badge_color_bg",
+        "label": "Custom Badge Color"
```
Comment (heythisisnate) [inline config/settings_schema.json:1352]: remove

File: sections/header.liquid [hunk 8]
```diff
@@ -246,7 +246,13 @@
     {%- endif -%}
 
     <div class="header__icons{% if section.settings.enable_country_selector or section.settings.enable_language_selector %} header__icons--localization header__localization{% endif %}">
-      <a class="shop-btn" href="/collections/shop-now">Shop Now</a>
+      {% assign shopLink = '/collections/shop-now' %}
+      
+      {% if customer.tags contains 'wholesale' %}
+        {% assign shopLink = '/collections/wholesale' %}
```
Comment (heythisisnate) [inline sections/header.liquid:252]: update to `/collections/partners-and-resellers`

File: snippets/card-product.liquid [hunk 9]
```diff
@@ -129,10 +129,15 @@
             {%- elsif card_product.compare_at_price > card_product.price and card_product.available -%}
               <span
                 id="NoMediaStandardBadge-{{ section_id }}-{{ card_product.id }}"
-                class="badge badge--bottom-left color-{{ settings.sale_badge_color_scheme }}"
+                class="badge {% if card_product.type == 'Wholesale' %}custom-badge{% endif %} badge--bottom-left color-{{ settings.sale_badge_color_scheme }}"
+                {% if card_product.type == 'Wholesale' %}style="--color-background: {{ settings.custom_badge_color_bg.red }}, {{ settings.custom_badge_color_bg.green }}, {{ settings.custom_badge_color_bg.blue }};"{% endif %}
```
Comment (heythisisnate) [inline snippets/card-product.liquid:133]: remove custom badge colors. just use sale badge styles

File: snippets/card-product.liquid [hunk 10]
```diff
@@ -326,9 +331,14 @@
           {%- elsif card_product.compare_at_price > card_product.price and card_product.available -%}
             <span
               id="Badge-{{ section_id }}-{{ card_product.id }}"
-              class="badge badge--bottom-left color-{{ settings.sale_badge_color_scheme }}"
+              class="badge badge--bottom-left {% if card_product.type == 'Wholesale' and customer.tags contains 'wholesale' %}custom-badge{% endif %} color-{{ settings.sale_badge_color_scheme }}"
+              {% if card_product.type == 'Wholesale' and customer.tags contains 'wholesale' %}style="--color-background: {{ settings.custom_badge_color_bg.red }}, {{ settings.custom_badge_color_bg.green }}, {{ settings.custom_badge_color_bg.blue }};"{% endif %}
```
Comment (heythisisnate) [inline snippets/card-product.liquid:335]: remove custom badge colors, use sale badge styles

File: snippets/header-drawer.liquid [hunk 11]
```diff
@@ -109,7 +109,14 @@
                 </li>
               {%- endfor -%}
               <li>
-                <a href="/collections/shop-now" class="menu-drawer__menu-item list-menu__item link link--text focus-inset">
+        
+                {% if customer.tags contains 'wholesale' %}
+                  {% assign shopLink = '/collections/wholesale' %}
```
Comment (heythisisnate) [inline snippets/header-drawer.liquid:114]: update link here too

File: snippets/price.liquid [hunk 12]
```diff
@@ -84,8 +84,12 @@
     </small>
   </div>
   {%- if show_badges -%}
-    <span class="badge price__badge-sale color-{{ settings.sale_badge_color_scheme }}">
-      {{ 'products.product.on_sale' | t }}
+    <span class="badge price__badge-sale color-{{ settings.sale_badge_color_scheme }} {% if product.type == 'Wholesale' and customer.tags contains 'wholesale' %}custom-badge{% endif %}" {% if product.type == 'Wholesale' and customer.tags contains 'wholesale' %}style="--color-background: {{ settings.custom_badge_color_bg.red }}, {{ settings.custom_badge_color_bg.green }}, {{ settings.custom_badge_color_bg.blue }};"{% endif %}>
```
Comment (heythisisnate) [inline snippets/price.liquid:87]: remove custom badge colors, just use sale badge colors

File: templates/collection.alarm-kits-sale.json
Comment (heythisisnate) [inline templates/collection.alarm-kits-sale.json:1]: Rename this to `collections.partners-and-resellers.json`

**Review by heythisisnate — CHANGES_REQUESTED** [review]

File: snippets/card-product.liquid (hunk 10 above)
Comment (chrisdavery) [reply snippets/card-product.liquid:335]: @heythisisnate we need these custom color pickers for badges, Dawn Colors are limited

image

image

Comment (chrisdavery) [reply snippets/card-product.liquid:335]: I should have replied under here sorry 
image

File: snippets/header-drawer.liquid (hunk 11 above)
Comment (heythisisnate) [reply snippets/header-drawer.liquid:114]: ok but there should only be two badge colors: "sale" and "sold out" 
parter price and sale is really the same badge (different text)

Comment (heythisisnate) [reply snippets/header-drawer.liquid]: I said no to the red color. Only two colors. Partner and Sale are both the green color.

File: snippets/locksmith-variables.liquid
Comment (heythisisnate) [inline snippets/locksmith-variables.liquid:1]: can you exclude this from the commit?

File: assets/konnected.css [hunk 13]
```diff
@@ -855,6 +855,49 @@ iframe#bwpFrame {
...
+.product .price--sold-out .price__badge-sold-out {
+  background: rgb(var(--color-background)) !important;
+}
+
+.price--on-sale .price__sale.swap-flex {
+  display: flex !important;
+  align-items: center;
+  flex-direction: row-reverse !important;
+  gap: 0.5rem;
+}
```
Comment (heythisisnate) [inline assets/konnected.css:888]: 😮  I never saw `flex-direction` property before!

File: templates/index.json
Comment (heythisisnate) [inline templates/index.json:1]: hmm is this change related?

**Review by heythisisnate — CHANGES_REQUESTED** [review]
much cleaner!
a few change requests in comments

File: templates/index.json
Comment (chrisdavery) [reply templates/index.json:1]: Yep. we need this

**Review by heythisisnate — APPROVED** [review]
